### ISSUE TITLE
H6053 - Flow Light Bar

### ISSUE THREAD

**Issue by christianonpele** [bwp91/homebridge-govee#89]
[Flow Light Bar] is not currently supported, do create a Github issue with the following info:

  device: 'F2:C1:A4:C1:38:0A:46:62',
  deviceName: 'Flow Light Bar',
  model: 'H6053',
  httpInfo: {
    device: 'F2:C1:A4:C1:38:0A:46:62',
    sku: 'H6053',
    spec: '',
    versionHard: '1.00.00',
    versionSoft: '1.01.01',
    deviceName: 'Flow Light Bar',
    deviceExt: {
      deviceSettings: '{"address":"A4:C1:38:0A:46:62","bleName":"ihoment_H6053_4662","pactType":1,"pactCode":1,"ic":0,"sku":"H6053","device":"F2:C1:A4:C1:38:0A:46:62","deviceName":"Flow Light Bar","versionHard":"1.00.00","versionSoft":"1.01.01"}',
      lastDeviceData: '{"online":false}',
      extResources: '{"skuUrl":"https://d1f2504ijhdyjw.cloudfront.net/sku-img/9e47ec098d87e1356621b0cbcd703428-add_list_type_device_6053.png","headOnImg":"https://d1f2504ijhdyjw.cloudfront.net/sku-img/541365e0fdfe06de14f074f0da8f64c4-new_light_title_6053_on.png","headOffImg":"http://s3.amazonaws.com/govee-public/sku-img/c55736f338216464a6b16782be73b2e0-new_light_title_6053_off.png","ext":"","ic":0}'
    },
    pactType: 1,
    pactCode: 1,
    goodsType: 23
  },
  isAPIDevice: false

https://fccid.io/2AQA6-H6053

Thank you!

**Comment by bwp91** [comment]
Hi @christianonpele 

Looks like this is a bluetooth only device.

Out of interest do you have any other models that you have gotten bluetooth working with this plugin?

**Comment by christianonpele** [comment]
EDIT: I think H6199 is connected via WiFi 

hi @bwp91 

yes I got a Govee Immersion (H6199) configured via your nice plugin.

[7/24/2021, 4:42:14 PM] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[7/24/2021, 4:42:19 PM] [HB Supervisor] Restarting Homebridge...
[7/24/2021, 4:42:19 PM] [HB Supervisor] Starting Homebridge with extra flags: -I -P /homebridge/node_modules -D
[7/24/2021, 4:42:20 PM] [HB Supervisor] Started Homebridge v1.3.4 with PID: 9523
Initializing HAP-NodeJS v0.9.4...
[7/24/2021, 4:42:25 PM] ---
[7/24/2021, 4:42:25 PM] Loaded plugin: homebridge-govee@3.7.0
[7/24/2021, 4:42:25 PM] Registering platform 'homebridge-govee.Govee'
[7/24/2021, 4:42:25 PM] ---
[7/24/2021, 4:42:27 PM] Loaded plugin: homebridge-config-ui-x@4.41.1
[7/24/2021, 4:42:27 PM] Registering platform 'homebridge-config-ui-x.config'
[7/24/2021, 4:42:27 PM] ---

Starting to advertise 'Homebridge 7FC4 C32E' using ciao backend!
[7/24/2021, 4:42:27 PM] Homebridge v1.3.4 (Homebridge 7FC4) is running on port 51396.
Initializing HAP-NodeJS v0.9.4...
[7/24/2021, 4:42:28 PM] [Govee] Launched child bridge with PID 9551
[7/24/2021, 4:42:29 PM] Registering platform 'homebridge-govee.Govee'
[7/24/2021, 4:42:29 PM] [Govee] Loaded homebridge-govee v3.7.0 child bridge successfully
[7/24/2021, 4:42:29 PM] Loaded 1 cached accessories from cachedAccessories.0E3D288A384C.
[7/24/2021, 4:42:29 PM] [Govee] Initialising plugin v3.7.0 | Node v14.17.3 | HB v1.3.4...
Preparing Advertiser for 'Govee F0CF' using ciao backend!
[7/24/2021, 4:42:29 PM] [Govee] Plugin initialised. Setting up accessories....
Starting to advertise 'Govee F0CF' using ciao backend!
[7/24/2021, 4:42:29 PM] Homebridge v1.3.4 (Govee) is running on port 46617.
[7/24/2021, 4:42:33 PM] [Govee] [Govee Immersion] initialising with options {"adaptiveLightingShift":0,"brightnessStep":1,"enableAWS":"disabled","enableBT":"disabled","logging":"debug"}.
[7/24/2021, 4:42:33 PM] [Govee] [Govee Immersion] initialised with id [70:4F:A4:C1:38:4C:60:38].
[7/24/2021, 4:42:33 PM] [Govee] [Flow Light Bar] is not currently supported, do create a Github issue with the following info:
[7/24/2021, 4:42:33 PM] [Govee] {
  device: 'F2:C1:A4:C1:38:0A:46:62',
  deviceName: 'Flow Light Bar',
  model: 'H6053',
  httpInfo: {
    device: 'F2:C1:A4:C1:38:0A:46:62',
    sku: 'H6053',
    spec: '',
    versionHard: '1.00.00',
    versionSoft: '1.01.01',
    deviceName: 'Flow Light Bar',
    deviceExt: {
      deviceSettings: '{"address":"A4:C1:38:0A:46:62","bleName":"ihoment_H6053_4662","pactType":1,"pactCode":1,"ic":0,"sku":"H6053","device":"F2:C1:A4:C1:38:0A:46:62","deviceName":"Flow Light Bar","versionHard":"1.00.00","versionSoft":"1.01.01"}',
      lastDeviceData: '{"online":false}',
      extResources: '{"skuUrl":"https://d1f2504ijhdyjw.cloudfront.net/sku-img/9e47ec098d87e1356621b0cbcd703428-add_list_type_device_6053.png","headOnImg":"https://d1f2504ijhdyjw.cloudfront.net/sku-img/541365e0fdfe06de14f074f0da8f64c4-new_light_title_6053_on.png","headOffImg":"http://s3.amazonaws.com/govee-public/sku-img/c55736f338216464a6b16782be73b2e0-new_light_title_6053_off.png","ext":"","ic":0}'
    },
    pactType: 1,
    pactCode: 1,
    goodsType: 23
  },
  isAPIDevice: false
}
[7/24/2021, 4:42:33 PM] [Govee] ✓ Setup complete. Thanks for using this plugin, I hope you find it helpful!
[7/24/2021, 4:42:33 PM] [Govee] [noble] stateChange: poweredOn.
[7/24/2021, 4:42:34 PM] [Govee] [Govee Immersion] [API] receiving update [{"source":"API","online":"true","powerState":"off","brightness":85,"color":{"r":255,"b":0,"g":0}}].

Have a nice day!

**Comment by christianonpele** [comment]
> 
> Looks like this is a bluetooth only device.
> 

Yep. the H6053 is BT only.

![IMG_7783](https://user-images.githubusercontent.com/19741754/126864826-a10e7584-2523-407c-a4e1-f2fd8343870b.PNG)

**Comment by bwp91** [comment]
I can easily add this model into the plugin bluetooth supported devices, but it can be a little more difficult on your homebridge instance actually getting the bluetooth to work.

I see you don’t have bluetooth enabled for your other device 

```
[7/24/2021, 4:42:33 PM] [Govee] [Govee Immersion] initialising with options {"adaptiveLightingShift":0,"brightnessStep":1,"enableAWS":"disabled","enableBT":"disabled","logging":"debug"}.
```

Are you running homebridge on a raspberry pi?

**Comment by christianonpele** [comment]
Dear @bwp91,

yes I tried to get the BT working HB. I'm having HB running on as docker container in HA:

https://community.home-assistant.io/t/tutorial-add-homebridge-to-hass-io-using-portainer/140423

Host sys is windows. There is a BT device successfully working in HA and another BT device idle at this VM. Unfortunately, my Linux knowledge is very limited. For me personal its not a big deal to move HB into another environment if BT can't established in that docker. 

Thank you for advice!

**Comment by bwp91** [comment]
I've just released a new beta version, v3.7.1-beta.0, which enables support for this model.

You'll need to:

* update to the beta version
* create an entry in the plugin configuration with the device id for your lights `F2:C1:A4:C1:38:0A:46:62` and enable bluetooth control
* there's also a couple of extra steps needed to get the bluetooth working on windows:

    https://github.com/abandonware/noble#windows

**Comment by fraherjr90** [comment]
I’m having a similar issue like this one, but my device is BT only H6055

[24/07/2021, 18:16:18] [Govee] [Table light] is not currently supported, do create a Github issue with the following info:
[24/07/2021, 18:16:19] [Govee] {
  device: '1C:88:CE:32:32:36:0E:1F',
  deviceName: 'Table light',
  model: 'H6055',
  httpInfo: {
    device: '1C:88:CE:32:32:36:0E:1F',
    sku: 'H6055',
    spec: '',
    versionHard: '3.02.01',
    versionSoft: '1.01.03',
    deviceName: 'Table light',
    deviceExt: {
      deviceSettings: '{"address":"CE:32:32:36:0E:1F","bleName":"Govee_H6055_0E1F","pactType":1,"pactCode":1,"ic":0,"secretCode":"Hw42MjLOYFU=","sku":"H6055","device":"1C:88:CE:32:32:36:0E:1F","deviceName":"Table light","versionHard":"3.02.01","versionSoft":"1.01.03"}',
      lastDeviceData: '{"online":false}',
      extResources: '{"skuUrl":"https://d1f2504ijhdyjw.cloudfront.net/sku-img/a323874176c3a1feebe37caf7268c9fe-add_list_type_device_6055.png","headOnImg":"https://d1f2504ijhdyjw.cloudfront.net/sku-img/0a61355a904e4e43d518c933c667a460-new_light_title_6055_on.png","headOffImg":"https://d1f2504ijhdyjw.cloudfront.net/sku-img/2db0d47f095b331e3e2b69b8d06d1e6d-new_light_title_6055_off.png","ext":"","ic":0}'
    },
    pactType: 1,
    pactCode: 1,
    goodsType: 42
  },
  isAPIDevice: false
}

**Comment by bwp91** [comment]
@fraherjr90 

These messages are there since bluetooth support is now available on some systems which can control bluetooth only lights.

I need to add the models manually, in your case the H6055 is not yet on any plugin supported list.

You can also hide this message by creating an entry for this device in the 'Light Devices' part of the config, with a label, device id `1C:88:CE:32:32:36:0E:1F` and ticking the 'Ignore From HomeKit' box.

**Comment by christianonpele** [comment]
> I've just released a new beta version, v3.7.1-beta.0, which enables support for this model.
> 
![Bildschirmfoto 2021-07-27 um 16 53 28](https://user-images.githubusercontent.com/19741754/127134913-dec40dde-5222-4b8d-bd2d-5d14580268c5.jpg)

Thank you so much bro! That worked for me.
